integrated_channels/integrated_channel/migrations/0004_catalogtransmissionaudit_channel.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('integrated_channel', '0003_catalogtransmissionaudit_learnerdatatransmissionaudit'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='catalogtransmissionaudit',
+            name='channel',
+            field=models.CharField(default='SAP', max_length=30),
```

Comment:
Good point and yeah I'm not sure what it should default to.

And I think you're right that we should make it a choice out of existing channels, but at the same time this wouldn't be visible in the admin, so it wouldn't be that useful. What do you think we should do? cc @douglashall

Reply:
Will migrations not work if we don't define a default?

Reply:
I realized there's an option `preserve_default` for `AddField` that we can set to `False` and have this as a one-off for all existing rows. The one-off default can be `'SAP'` because all existing rows are for that, and then this can carry on without one; since the field is non-nullable, new rows will require their own channel, i.e. `'DEGREED'`.

Reply:
@UmanShahzad 

> ... there's an option `preserve_default` for `AddField` that we can set to `False` and have this as a one-off for all existing rows.

\#TIL